### PR TITLE
chore(tracing): add span info for Condition evaluation

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
Add span info to traces for Condition evaluation results. See the screenshots for examples.

Screenshot 2023-12-19 at 3 04 30 PM
Screenshot 2023-12-19 at 3 04 30 PM

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: internal/condition/condition.go [hunk 1]
```diff
@@ -176,12 +180,18 @@ func (e *EvaluableCondition) CastContextToTypedParameters(contextMap map[string]
 	return converted, nil
 }
 
-// Evaluate evalutes the provided CEL condition expression with a CEL environment
+// EvaluateWithContext evalutes the provided CEL condition expression with a CEL environment
```
Comment (miparnisari) [inline internal/condition/condition.go:183]:
```suggestion
// EvaluateWithContext evaluates the provided CEL condition expression with a CEL environment
```

File: internal/condition/condition.go [hunk 2]
```diff
@@ -264,6 +274,13 @@ func (e *EvaluableCondition) Evaluate(contextMaps ...map[string]*structpb.Value)
 	}, nil
 }
 
+// Evaluate evalutes the provided CEL condition expression with a CEL environment
+// constructed from the condition's parameter type definitions and using the
+// context provided. See EvaluateWithContext above for more info.
+func (e *EvaluableCondition) Evaluate(contextMaps ...map[string]*structpb.Value) (EvaluationResult, error) {
```
Comment (miparnisari) [inline internal/condition/condition.go:280]:
```suggestion
// Evaluate evaluates the provided CEL condition expression with a CEL environment
// constructed from the condition's parameter type definitions and using a new
// context. See EvaluateWithContext above for more info.
func (e *EvaluableCondition) Evaluate(contextMaps ...map[string]*structpb.Value) (EvaluationResult, error) {
```

**Review by miparnisari — DISMISSED** [review]

File: internal/condition/eval/eval.go [hunk 3]
```diff
@@ -27,18 +34,30 @@ var conditionEvaluationCostHistogram = promauto.NewHistogram(prometheus.Histogra
...
 ) (*condition.EvaluationResult, error) {
+	ctx, span := tracer.Start(ctx, "EvaluateTupleCondition")
+	defer span.End()
+
+	span.SetAttributes(attribute.String("tuple_key", tupleKey.String()))
+
 	tupleCondition := tupleKey.GetCondition()
 	conditionName := tupleCondition.GetName()
 	if conditionName != "" {
+		span.SetAttributes(attribute.String("condition_name", conditionName))
```
Comment (miparnisari) [inline internal/condition/eval/eval.go:50]: This one is arguably redundant since the `tuple_key` attribute already logs the condition name

Comment (jon-whit) [reply internal/condition/eval/eval.go:50]: I debated that. I will remove it in favor of the tuple_key.

**Review by miparnisari — APPROVED** [review]

**Pull request merged**